hudi-common/src/main/java/org/apache/hudi/common/table/log/block/HoodieParquetDataBlock.java

```diff
@@ -143,6 +143,7 @@ protected <T> ClosableIterator<HoodieRecord<T>> readRecordsFromBlockPayload(Hood
     //       is appropriately carried over
     Configuration inlineConf = new Configuration(blockContentLoc.getHadoopConf());
     inlineConf.set("fs." + InLineFileSystem.SCHEME + ".impl", InLineFileSystem.class.getName());
+    inlineConf.set("spark.sql.parquet.writeLegacyFormat", "true");
```

Comment:
Please remove this one as well

Reply:
@wzx140 we need to review all the changes holistically to make sure we're not landing any unilateral config changes like that along w/ RFC-46